backend/controllers/notes.ts

```diff
@@ -28,34 +19,20 @@ notesRouter.get("/:id", async (request, response) => {
 	}
 });
 
-notesRouter.post("/", async (request, response) => {
+notesRouter.post("/", middleware.validateToken, async (request: ExtendedRequest, response) => {
 	const body = request.body;
-	const token = getTokenFrom(request);
-	if (process.env.SECRET && token) {
-		// CHECK
-		const decodedToken = jwt.verify(
-			token,
-			process.env.SECRET
-		) as jwt.JwtPayload; // Check again
-		if (!decodedToken.id) {
-			return response.status(401).json({ error: "token invalid" });
-		}
-
-		const user = await User.findById(decodedToken.id);
-
-		if (user) {
-			const note = new Note({
-				content: body.content,
-				important: body.important || false,
-				user: user.id,
-			});
-
-			const savedNote = await note.save();
-			user.notes = user.notes.concat(savedNote._id);
-			await user.save();
-
-			response.status(201).json(savedNote);
-		}
+	const user = await User.findById(request.userId);
+	if (user) {
+		const note = new Note({
+			content: body.content,
+			important: body.important || false,
+			user: user.id,
+		});
+
+		const savedNote = await note.save();
+		await user.save();
```

Comment:
We didn't modify any fields for `user`. Why do we need to call `user.save()` here?